File: practice/algorithms/sorting/counting_sort_2/solution.scala
import scala.io.Source

object Solution{
  def countingSort(arr: Array[Int]): Array[Int] =  {
    val countsOfNumbers = arr.sorted.groupBy(i => i).map(a => (a._1, a._2.length))

    (0 until 100).flatMap { i =>
      val n = countsOfNumbers.getOrElse(i, 0)
      (0 until n).map(j => i)
    }.toArray
  }

  def main(args: Array[String]) {
    val lines = Source.stdin.getLines().toList
    val arr = lines.tail.head.split(" ").map(_.toInt)
    val result = countingSort(arr)
    println (result.mkString(" "))
  }
}

File: practice/algorithms/sorting/counting_sort_2/solution.js
'use strict';

const counting = (arr) => {
    const counts = []
    for(let i = 0; i < 100; i++) {
        counts[i] = 0;
    }
    for(let i = 0; i < arr.length; i++) {
        counts[arr[i]]++;
    }

    let result = "";
    for (let i = 0; i < counts.length; i++) {
        for(let j = 0; j < counts[i]; j++) {
            result += (i + " ");
        }
    }
    console.log(result);
}

const processData = input => {
    const lines = input.split('\n');
    const arr  = lines[1].split(' ').map(i => parseInt(i));
    counting(arr);
};

process.stdin.resume();
process.stdin.setEncoding("ascii");

let _input = "";
process.stdin.on("data", input => _input += input);
process.stdin.on("end", () => processData(_input));